doc/nrf/libraries/emds/emds.rst

```diff
@@ -0,0 +1,52 @@
+
+Overview
+********
+The :kconfig:`CONFIG_EMDS` enables the emergency data storage. The application must initialize by using :ref:`emds_init`, the pre-allocated flash area used to store data in.
+The :kconfig:`CONFIG_EMDS_SECTOR_COUNT` defines how many sectors that should be used to store data. The space set aside shall be large than all the data stored. The larger value will increase the flash spaced used, but will reduce the wear of the flash area.
+This has to be balanced between amount of data stored, flash available and how of it will call the :ref:`emds_store`. It should not be necessary to have a large value, as there should be a limited set of data stored this way, as time is of essence here.
+
+The memory location that is going to be stored need to be added on initialization. All memory areas need to be provided through the :ref:`emds_entry_add` and :ref:`EMDS_STATIC_ENTRY_DEFINE`.
+When adding a memory area that should be stored when the emergency data storage is triggered it is important the each entry has it's own unique id that can be recreated after a reboot.
+
+The :ref:`emds_entry_load` will restore the data into the memory areas from the flash. This has to be done after all the memory areas has been added.
+
+After the data has been restored the application shall run a :ref:`emds_prepare`. This to get the pre allocated flash area prepared for receiving data, when the emergency data storage is needed. Available flash area must be larger than all data stored in the emergency data storage. If multiple store has been run and the flash area that is left is smaller than the needed data size, the flash area will be automatically erased to increase the empty flash area available.
+
+The storage will be done in a deterministic time so it will be possible know how long it takes to stored all the registered data. However this is dependent on the chip so to measure time is important and :kconfig:`EMDS_FLASH_TIME_BASE_OVERHEAD_US`, :kconfig:`EMDS_FLASH_TIME_ENTRY_OVERHEAD_US` and :kconfig:`EMDS_FLASH_TIME_BASE_OVERHEAD_US` can be configured. It is important when :kconfig:`EMDS_FLASH_TIME_BASE_OVERHEAD_US` is added time the worst time is considered, that including time for erase cycles when doing garbage collection.
+This will also prevent wear and tear of flash as data that are changing often are stored in memory, but can be stored in flash and be restored after reboot.
+
+The application need to call :ref:`emds_store` to store all the RAM areas provided. This can be only be done once before it need to run :ref:`emds_prepare` again. When the data storage is complete a callback will be called if provided in :ref:`emds_init`. This to notify to the application that the data storage is complete. This can be used to reboot the cpu or any other function that maybe needed.
+It will also make the :ref:`emds_is_ready` function call return false, as it is no longer can guaranty that the storage will fit in the amount of flash space left.
```

Comment:
```suggestion
Before shutdown the application must call :ref:`emds_store` to store all the RAM areas provided. This can be only be done once before it need to run :ref:`emds_prepare` again. When the data storage is complete a callback will be called if provided in :ref:`emds_init`. This callback notifies the application that the data storage is complete, and can be used to reboot the cpu or execute other function that is needed.
After completion of :ref:`emds_store` the :ref:`emds_is_ready` function call will return false, since it can no longer guarantee that the data will fit in the flash area.
```